provider.go

```diff
@@ -77,6 +83,63 @@ func readJWTToken(path string) (string, error) {
 	return string(bytes.TrimSpace(data)), nil
 }
 
+func splitSecretPath(secretPath string) (string, string) {
+	s := regexp.MustCompile("/+").Split(secretPath, 3)
+	return s[1], s[2]
+}
+
+func (p *Provider) getMountInfo(mountName string, token string) (string, string, error) {
+	client, err := p.createHTTPClient()
+	if err != nil {
+		return "", "", err
+	}
+
+	addr := p.VaultAddress + "/v1/sys/mounts"
+	req, err := http.NewRequest(http.MethodGet, addr, nil)
+	// Set vault token.
+	req.Header.Set("X-Vault-Token", token)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "couldn't generate request")
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "couldn't get sys mounts")
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		var b bytes.Buffer
+		io.Copy(&b, resp.Body)
+		return "", "", fmt.Errorf("failed to get successful response: %#v, %s",
+			resp, b.String())
+	}
+
+	var mount struct {
+		Data map[string]Mount `json:"data"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&mount); err != nil {
+		fmt.Printf("error: %s", err)
+	}
+
+	return mount.Data[mountName+"/"].Type, mount.Data[mountName+"/"].Options["version"], nil
+}
+
+func generateSecretEndpoint(vaultAddress string, secretMountType string, secretMountVersion string, secretPrefix string, secretSuffix string, secretVersion string) string {
```

Comment:
I'd suggest returning an error here, in particular explicit errors if `secretMountType != "kv"` or the version isn't 1 or 2.